### PR TITLE
refactor(ts): remove implicit any

### PR DESCRIPTION
**Pull request by Haroenv**
- adds missing @types devDependency
- adds missing annotations 
- rewrite the types for the answers connector

### CODE REVIEW

**Comment by Haroenv** [comment]
Actually not ready for review, I'll wait until #4785 is fully merged, in case there's some conflicts, I'd rather them be in this PR

**Review by shortcuts — COMMENTED** [review]
Cool PR! Just a few questions, I'll do an other review after they are answered :D

File: src/widgets/refinement-list/refinement-list.tsx
Comment (shortcuts) [reply src/widgets/refinement-list/refinement-list.tsx:168]: ^ this might be auto-fixed after the master merge?

File: src/components/Slider/Rheostat.tsx
Comment (Haroenv) [reply src/components/Slider/Rheostat.tsx:311]: yes, for the time being there's no checks, so clearly certain things that we mark as optional get defaulted at a certain step, but putting the ! here indicates the possible problem in the future, I didn't want to change code in this pr

File: src/widgets/stats/stats.tsx
Comment (Haroenv) [reply src/widgets/stats/stats.tsx:95]: the remaining ones don't use prepareTemplateProps now that master is merged :)

File: src/widgets/refinement-list/refinement-list.tsx
Comment (Haroenv) [reply src/widgets/refinement-list/refinement-list.tsx:168]: it's in the file (same for menu, hierarchical menu, rating menu) because the refinement list component is shared

File: src/components/Breadcrumb/Breadcrumb.tsx
Comment (tkrugg) [inline src/components/Breadcrumb/Breadcrumb.tsx:21]: Open question: wouldn't it be better to have a specific type here. Using the connector type leaks the internal detail this `BreadcrumpProps.items` is the same as `BreadcrumbConnectorParamsItem`, when the widget user doesn't really have to know about it.

File: src/components/Breadcrumb/Breadcrumb.tsx
Comment (tkrugg) [inline src/components/Breadcrumb/Breadcrumb.tsx:24]: Not sure why we need to handle`null` here. I don't know in which case is this value `null` ?

File: src/components/Slider/Slider.tsx
Comment (tkrugg) [inline src/components/Slider/Slider.tsx:26]: are we happy using the widget type here?
Currently widgets depends on component, not the other way around.

File: src/components/Slider/Slider.tsx
Comment (tkrugg) [inline src/components/Slider/Slider.tsx:80]: isn't `'aria-valuenow'` optional? I think casting as unknown hide the fact this value can be undefined but I'm not yet sure how we can avoid that.

File: src/connectors/numeric-menu/connectNumericMenu.ts
Comment (tkrugg) [inline src/connectors/numeric-menu/connectNumericMenu.ts:139]: TIL !

**Review by tkrugg — APPROVED** [review]

File: src/components/Slider/Slider.tsx
Comment (Haroenv) [reply src/components/Slider/Slider.tsx:26]: There's other components where that's the case. It's true it's a little coupled at the moment, but all components for example rely on the widget for the css classes (instead of the other way round), so we can fix that some other time (add a lint rule to avoid depending too circular), fixing this one instance is insignificant.

File: src/components/Slider/Slider.tsx
Comment (Haroenv) [reply src/components/Slider/Slider.tsx:80]:
```suggestion
        parseFloat((props['aria-valuenow'] as unknown) as (string | undefined)) * 100
```

is this better?

File: src/components/Breadcrumb/Breadcrumb.tsx
Comment (Haroenv) [reply src/components/Breadcrumb/Breadcrumb.tsx:21]: This is the component, so it's not public anyway :) But I import flow this can better be solved globally

File: src/components/Slider/Slider.tsx
Comment (Haroenv) [reply src/components/Slider/Slider.tsx:80]: parseFloat doesn't accept undefined, so it will be for some other time when we figure out why it even needs to be parsed as a float if it's already a number

File: src/components/Breadcrumb/Breadcrumb.tsx
Comment (Haroenv) [reply src/components/Breadcrumb/Breadcrumb.tsx:24]: https://github.com/algolia/instantsearch.js/blob/c57d7bbdd9451299b3af48ce6f894b012c24b9d5/src/connectors/breadcrumb/connectBreadcrumb.ts#L270

this is what makes it null, and I don't see how to avoid it (last item?)

Comment (tkrugg) [reply src/components/Breadcrumb/Breadcrumb.tsx:24]: or right! that's when the value has been selected! 👍  thanks for clarifying.

**Pull request merged**
